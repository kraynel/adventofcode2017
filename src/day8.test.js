const { largestRegister } = require("./day8");
const fs = require("fs");

describe("largestRegister", () => {
  it("should return the largest register", () => {
    expect(
      largestRegister(`b inc 5 if a > 1
a inc 1 if b < 5
c dec -10 if a >= 1
c inc -20 if c == 10`).max
    ).toBe(1);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day8.txt", "utf8", (err, data) => {
      expect(largestRegister(data).max).toMatchSnapshot();
      done();
    });
  });

  it("should return the largest register", () => {
    expect(
      largestRegister(`b inc 5 if a > 1
a inc 1 if b < 5
c dec -10 if a >= 1
c inc -20 if c == 10`).allTimeMax
    ).toBe(10);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day8.txt", "utf8", (err, data) => {
      expect(largestRegister(data).allTimeMax).toMatchSnapshot();
      done();
    });
  });
});
