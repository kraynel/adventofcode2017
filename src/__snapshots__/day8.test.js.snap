// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`largestRegister should get the value 1`] = `5102`;

exports[`largestRegister should get the value 2`] = `6056`;
